src/pkg/packager/layout/package.go

```diff
@@ -124,9 +125,136 @@ func (p *PackageLayout) ContainsSBOM() bool {
 	return !helpers.InvalidPath(filepath.Join(p.dirPath, SBOMTar))
 }
 
-// SignPackage signs the zarf package
-func (p *PackageLayout) SignPackage(signingKeyPath, signingKeyPassword string) error {
-	return signPackage(p.dirPath, signingKeyPath, signingKeyPassword)
+// SignPackage signs the zarf package using cosign with the provided options.
+// If the options do not indicate signing should be performed (no key material configured),
+// this is a no-op and returns nil.
+//
+// This function:
+// 1. Updates Pkg.Build.Signed = true in memory
+// 2. Writes the updated zarf.yaml (with signed:true) to a temporary file
+// 3. Signs the temporary file
+// 4. If signing succeeds, replaces the actual zarf.yaml with the signed version
```

Comment:
Generally for doc comments on public functions I think we should highlight the outcomes caused by the functions rather than the implementation details. To me, uses a temporary file is an implementation detail.